packages/hooks/src/useLoggedStateLoginAndLogout.ts

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useRef, useState } from 'react';
+import Cookies from 'js-cookie';
+
+import { requestOidcAuthentication } from '@deriv-com/auth-client';
+import { useStore } from '@deriv/stores';
+import { isSafariBrowser } from '@deriv/shared';
+/**
+ * Handles silent login and single logout logic for OAuth2.
+ *
+ * @param {{
+ *   is_client_store_initialized: boolean; // Whether the client store has been initialized
+ *   isOAuth2Enabled: boolean; // Whether OAuth2 feature is enabled
+ *   oAuthLogout: () => Promise<void>; // Function to handle OAuth2 logout
+ * }} params - The arguments required for silent login and logout management
+ */
+const useSilentLoginAndLogout = ({
+    is_client_store_initialized,
+    isOAuth2Enabled,
+    oAuthLogout,
+}: {
+    is_client_store_initialized: boolean;
+    isOAuth2Enabled: boolean;
+    oAuthLogout: () => Promise<void>;
+}) => {
+    const loggedState = Cookies.get('logged_state');
+    const [is_single_logging_in, setIsSingleLoggingIn] = useState(false);
+
+    const { client } = useStore();
+    const clientAccounts = JSON.parse(localStorage.getItem('client.accounts') || '{}');
+    const clientTokens = JSON.parse(localStorage.getItem('config.tokens') || '{}');
+    const isClientAccountsPopulated = Object.keys(clientAccounts).length > 0;
+    const isClientTokensPopulated = Object.keys(clientTokens).length > 0;
+    const isSilentLoginExcluded = ['callback', 'silent-callback', 'front-channel', 'endpoint'].some(path =>
+        window.location.pathname.includes(path)
+    );
+
+    // state to manage and ensure OIDC callback functions are invoked once only
+    const isAuthenticating = useRef(false);
+    const isLoggingOut = useRef(false);
+    const { prevent_single_login } = client;
+
+    useEffect(() => {
+        const willEventuallySSO = loggedState === 'true' && !isClientAccountsPopulated && !isClientTokensPopulated;
+        if (willEventuallySSO && !isSilentLoginExcluded) {
+            setIsSingleLoggingIn(true);
+        } else {
+            setIsSingleLoggingIn(false);
+        }
```

Comment:
I moved the logic for showing the SSO loader to this hook (the hook that handles logged state)